app/code/core/Mage/Core/Model/Config.php

```diff
@@ -1411,109 +1416,145 @@ public function getBlockClassName($blockType)
+    public function getModelInstance($modelAlias, $constructArguments = [])
     {
-        $className = $this->getModelClassName($modelClass);
-        if (class_exists($className)) {
-            Varien_Profiler::start('CORE::create_object_of::' . $className);
-            $obj = new $className($constructArguments);
-            Varien_Profiler::stop('CORE::create_object_of::' . $className);
-            return $obj;
-        } else {
+        $className = $this->getModelClassName($modelAlias);
+        if (!class_exists($className)) {
             return false;
         }
+        Varien_Profiler::start('CORE::create_object_of::' . $className);
+        // phpcs:ignore Ecg.Classes.ObjectInstantiation.DirectInstantiation
+        $obj = new $className($constructArguments);
+        Varien_Profiler::stop('CORE::create_object_of::' . $className);
+        return $obj;
     }
 
     /**
-     * @param string $path
-     * @return bool
+     * Retrieve resource model class name
+     *
+     * @param string $modelAlias
+     * @return string|false
      */
-    public function getNodeClassInstance($path)
+    public function getResourceModelClassName($modelAlias)
     {
-        $config = Mage::getConfig()->getNode($path);
-        if (!$config) {
-            return false;
-        } else {
-            $className = $config->getClassName();
-            return new $className();
+        $factoryName = $this->_getResourceModelFactoryClassName($modelAlias);
+        if ($factoryName) {
+            return $this->getModelClassName($factoryName);
         }
+        return false;
     }
 
     /**
-     * Get resource model object by alias
+     * Retrieve resource model instance
      *
-     * @param   string $modelClass
-     * @param   array $constructArguments
+     * @param string $modelAlias
+     * @param array $constructArguments
      * @return Mage_Core_Model_Resource_Db_Collection_Abstract|false
      */
-    public function getResourceModelInstance($modelClass = '', $constructArguments = [])
+    public function getResourceModelInstance($modelAlias, $constructArguments = [])
     {
-        $factoryName = $this->_getResourceModelFactoryClassName($modelClass);
-        if (!$factoryName) {
+        $className = $this->getResourceModelClassName($modelAlias);
+        if ($className === false || !class_exists($className)) {
             return false;
         }
-        return $this->getModelInstance($factoryName, $constructArguments);
+        Varien_Profiler::start('CORE::create_object_of::' . $className);
+        // phpcs:ignore Ecg.Classes.ObjectInstantiation.DirectInstantiation
+        $obj = new $className($constructArguments);
+        Varien_Profiler::stop('CORE::create_object_of::' . $className);
+        return $obj;
+    }
+
+    /**
+     * Retrieve resource helper class name
+     */
+    public function getResourceHelperClassName(string $moduleAlias): string|false
```

Comment:
"string|false" union type is php8? Possible to change return type for new methods to null?

Reply:
I thought I caught all of those. Put in `@return` annotation. Best to leave it as `string|false` to match the other `getXxxClassName` methods.